compiler/transf.nim

```diff
@@ -563,7 +563,7 @@ proc putArgInto(arg: PNode, formal: PType): TPutArgInto =
       return paDirectMapping    # XXX really correct?
                                 # what if ``arg`` has side-effects?
   case arg.kind
-  of nkEmpty..nkNilLit:
+  of nkEmpty..nkNilLit, nkDotExpr:
```

Comment:
What if it's ``sideEffectHere().field``? Then we cannot use a direct mapping, can we?

Reply:
Yep, fixed